components/tools/OmeroWeb/omeroweb/webgateway/marshal.py

```diff
@@ -134,9 +134,18 @@ def imageMarshal(image, key=None):
 
     # big images
     tiles = image._re.requiresPixelsPyramid()
-    width, height = image._re.getTileSize()
-    levels = image._re.getResolutionLevels()
-    zoomLevelScaling = image.getZoomLevelScaling()
+    rv['tiles'] = tiles
+    if (tiles):
```

Comment:
I'm not sure. You'd have to go through all the code paths. This code is ancient and goes well, well back to before 2012 (4.4.0) unchanged. The `try` block you're referring to is as it was when I originally moved all of this marshalling out of `omeroweb.webgateway.views` as part of #313 in 2012 so that shape marshalling could be unit tested.

I would encourage this PR to be unit and/or integration tested as well. Extending `test/unit/test_marshal.py` would be perfect.